src/ldp/http/OriginalUrlExtractor.ts

```diff
@@ -37,7 +44,13 @@ export class OriginalUrlExtractor extends TargetExtractor {
 
     // URL object applies punycode encoding to domain
     const base = `${protocol}://${host}`;
-    const path = new URL(toCanonicalUriPath(url), base).href;
-    return { path };
+    const originalUrl = new URL(toCanonicalUriPath(url), base);
+
+    // Drop the query string if requested
+    if (!this.includeQueryString) {
+      originalUrl.search = '';
```

Comment:
It's actually funny this only pops up now because I always assumed the search string was ignored already. 😄